File: libanimations/src/main/java/com/pedro/libanimations/custom/VibrateXAnimation.java
package com.pedro.libanimations.custom;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.view.View;
import com.pedro.libanimations.base.BaseAnimation;

/**
 * Created by pedro on 13/10/16.
 */

public class VibrateXAnimation extends BaseAnimation {

  private AnimatorSet mAnimationSet;
  private int delay;
  private boolean state = false;
  private View v;

  public VibrateXAnimation(int delay){
    mAnimationSet = getmAnimationSet();
    this.delay = delay;
  }

  @Override
  public void startAnimation(View v) {
    this.v = v;
    ObjectAnimator animator = moveMore(v);
    animator.setStartDelay(delay);
    mAnimationSet.play(animator);
    mAnimationSet.start();
  }

  private ObjectAnimator moveMore(View v){
    ObjectAnimator animatorX = ObjectAnimator.ofFloat(v, "translationX", 3);
    animatorX.setDuration(20);
    state = true;
    return animatorX;
  }

  private ObjectAnimator moveLess(View v){
    ObjectAnimator animatorX = ObjectAnimator.ofFloat(v, "translationX", -3);
    animatorX.setDuration(20);
    state = false;
    return animatorX;
  }

  @Override
  public void cancelAnimation() {

  }

  @Override
  public void stopAnimation() {

  }

  @Override
  public void onFinish() {
    if(!state) mAnimationSet.play(moveMore(v));
    else  mAnimationSet.play(moveLess(v));
    mAnimationSet.start();
  }
}

File: libanimations/src/main/java/com/pedro/libanimations/custom/VibrateXAnimation.kt
package com.pedro.libanimations.custom

import android.animation.ObjectAnimator
import android.view.View
import com.pedro.libanimations.base.BaseAnimation

/**
 * Created by pedro on 13/10/16.
 */

class VibrateXAnimation(val delay: Int = 0) : BaseAnimation() {

    private var state = false
    private var v: View? = null

    override fun startAnimation(v: View) {
        this.v = v
        val animator = moveMore(v)
        animator.startDelay = delay.toLong()
        animationSet.play(animator)
        animationSet.start()
    }

    private fun moveMore(v: View?): ObjectAnimator {
        val animatorX = ObjectAnimator.ofFloat(v, "translationX", 3f)
        animatorX.duration = 20
        state = true
        return animatorX
    }

    private fun moveLess(v: View?): ObjectAnimator {
        val animatorX = ObjectAnimator.ofFloat(v, "translationX", -3f)
        animatorX.duration = 20
        state = false
        return animatorX
    }

    override fun cancelAnimation() {

    }

    override fun stopAnimation() {

    }

    override fun onFinish() {
        if (!state) animationSet.play(moveMore(v))
        else animationSet.play(moveLess(v))
        animationSet.start()
    }
}
